sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormTrainingAsyncClient.java

```diff
@@ -0,0 +1,292 @@
+        return beginTraining(fileSourceUrl, useLabelFile, false, null, null);
+    }
+
+    /**
+     * Create and train a custom model.
+     * <p>Models are trained using documents that are of the following content type -
+     * 'application/pdf', 'image/jpeg', 'image/png', 'image/tiff'.
+     * Other type of content is ignored.
+     * </p>
+     * <p>The service does not support cancellation of the long running operation and returns with an
+     * error message indicating absence of cancellation support.</p>
+     *
+     * @param fileSourceUrl source URL parameter that is either an externally accessible Azure
+     * storage blob container Uri (preferably a Shared Access Signature Uri).
+     * @param useLabelFile Boolean to specify the use of labeled files for training the model.
+     * @param includeSubFolders to indicate if sub folders within the set of prefix folders will
+     * also need to be included when searching for content to be preprocessed.
+     * @param filePrefix A case-sensitive prefix string to filter documents in the source path
+     * for training. For example, when using a Azure storage blob Uri, use the prefix to restrict
+     * sub folders for training.
+     * @param pollInterval Duration between each poll for the operation status. If none is specified, a default of
+     * 5 seconds is used.
+     *
+     * @return A {@link PollerFlux} that polls the extract receipt operation until it
+     * has completed, has failed, or has been cancelled.
+     */
+    public PollerFlux<OperationResult, CustomFormModel> beginTraining(String fileSourceUrl,
+        boolean useLabelFile, boolean includeSubFolders, String filePrefix, Duration pollInterval) {
+        Objects.requireNonNull(fileSourceUrl, "'fileSourceUrl' cannot be null.");
+        Objects.requireNonNull(useLabelFile, "'useLabelFile' cannot be null.");
+        final Duration interval = pollInterval != null ? pollInterval : Duration.ofSeconds(5);
+
+        return new PollerFlux<OperationResult, CustomFormModel>(
+            interval,
+            getTrainingActivationOperation(fileSourceUrl, includeSubFolders, filePrefix, useLabelFile),
+            createTrainingPollOperation(),
+            (activationResponse, context) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            fetchTrainingModelResultOperation());
+    }
+
+    /**
+     * Get detailed information for a specified custom model id.
+     *
+     * @param modelId Model identifier.
+     *
+     * @return The detailed information for the specified model.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CustomFormModel> getCustomModel(String modelId) {
+        return getCustomFormModelWithResponse(modelId).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Get detailed information for a specified custom model id with Http response
+     *
+     * @param modelId Model identifier.
```

Comment:
Update javadoc to specify that the model identifier is expected to be in UUID format. I would lean towards taking UUID as the input directly as the user anyway has to generate one from `UUID.randomUUID()`.

Reply:
It is expected as String for now as it allows the service team to give them an option to evolve to other string formats in the future.